cmake/Modules/CompilerConfig.cmake

```diff
@@ -90,16 +90,36 @@ else()
   endif()
 
   add_compile_options(
+    -Werror
     -Wextra
     -Wvla
+    -Wswitch
+    -Wno-error=switch
+    -Wformat
+    -Wformat-security
+    -Wunused-parameter
     -Wno-unused-function
     -Wno-missing-field-initializers
     -fno-strict-aliasing
     "$<$<COMPILE_LANGUAGE:C>:-Werror-implicit-function-declaration;-Wno-missing-braces>"
     "$<$<BOOL:${USE_LIBCXX}>:-stdlib=libc++>"
     "$<$<CONFIG:DEBUG>:-DDEBUG=1;-D_DEBUG=1>"
-    "$<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-fcolor-diagnostics>"
-    "$<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang>:-fcolor-diagnostics>")
+    "$<$<COMPILE_LANG_AND_ID:CXX,AppleClang,Clang>:-Wnull-conversion;-fcolor-diagnostics;-Wno-error=shorten-64-to-32>"
+    "$<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang>:-Wnull-conversion;-fcolor-diagnostics;-Wno-error=shorten-64-to-32>"
+    "$<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wconversion-null>")
+
+  if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # GCC 12.1.0 has a regression bug which trigger maybe-uninitialized warnings
+    # where there is not. (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105562)
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "12.1.0")
+      add_compile_options(-Wno-error=maybe-uninitialized)
+    endif()
```

Comment:
Is this still needed with 12.2.0 being out for a while now?

Reply:
Removed.

Reply:
Will be re-added require for Flatpak.

Reply:
They appear to use 12.1.0 on 22.08.